src/IndustrialRobot/IndustrialRobot/ViewModels/MainViewModel.cs

```diff
@@ -1,14 +1,97 @@
 ﻿using Caliburn.Micro;
 using System;
 using System.Collections.Generic;
+using System.IO.Ports;
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Shapes;
+using IndustrialRobot.Views;
+using GalaSoft.MvvmLight.Command;
+using IndustrialRobot.Views.Interfaces;
 
 namespace IndustrialRobot.ViewModels
 {
-    public class MainViewModel : Screen
+    public class MainViewModel : Window 
     {
+        public SerialPort serialPort = new SerialPort();
+        public ICommand StartButtonCommand { get; set; }        
+        public ICommand SettingsButtonCommand { get; set; }
+        public ICommand ExitButtonCommand { get; set; }
 
+        //public RelayCommand<ICloseable> CloseWindowCommand { get; private set; }
+
+        public MainViewModel()
+        {
+            StartButtonCommand = new RelayCommand(ExecuteStartButton, CanExecuteStartButton);
+            SettingsButtonCommand = new RelayCommand(ExecuteSettingsButton, CanExecuteSettingsButton);
+            ExitButtonCommand = new RelayCommand(ExecuteExitButton, CanExecuteExitButton);
+            //CloseWindowCommand = new RelayCommand<ICloseable>(CloseWindow);     
+        }
+        private bool CanExecuteStartButton(object parameter)
+        {
+            return true;
+        }
+
+        private void ExecuteStartButton(object parameter)
+        {
+            serialPort.Close();
+            try
+            {
+                if (serialPort.IsOpen == false)
+                {
+                    serialPort.PortName = Properties.Settings.Default.Port;
+                    serialPort.BaudRate = Properties.Settings.Default.BaudRate;
+                    serialPort.Parity = Properties.Settings.Default.Parity;
+                    serialPort.DataBits = Properties.Settings.Default.DataBits;
+                    serialPort.StopBits = Properties.Settings.Default.StopBits;
+                    serialPort.Open();
+                }
+                ControlPanelView controlPanelView = new ControlPanelView(this);
```

Comment:
to bedziemy musieli pozniej przerobic. Mam namysli dlczego przekazujesz this / MainViewModel do ControlPanelView?? One powinny byc zupelnie niezalezne. I jesli cos z MainViewModel bedziesz potrzebowal tam to musisz tylko przekazac konkretne rzecyz. Jelsi jest jakas grubsza zaleznosc, to tak naprawde bedzie to pewnie trzeba przeprojektowac. Na ten moment rzuc okiem dlaczego to przekazujesz i odpisz mi prosze w komencie..

Reply:
zmienione, na ten moment przekazuje sam SerialPort